### DISCUSSION TITLE
Should prompt always use UTF-8 even when the connected console may be using a different encoding?

### DISCUSSION THREAD

**Discussion by mfulton26** [denoland/deno#11773] [Q&A]
My question is prompted (pun intended) by [unicode - Deno: how to take care of accents when reading from windows terminal prompt? - Stack Overflow](https://stackoverflow.com/questions/68839705/deno-how-to-take-care-of-accents-when-reading-from-windows-terminal-prompt).

Should [neves](https://stackoverflow.com/users/10335/neves) not have run into an issue here? If [`prompt`](https://doc.deno.land/builtin/stable#prompt) were to detect and use whatever encoding the TTY is using (e.g. `windows-1252` instead of `utf-8`) then it would have worked with no problems. Would doing such cause some web standard compatibility/divergence issue or as it is isolated to Deno's terminal `window` it might be appropriate.

Thank you.

**Comment by paulobndes** [comment]
I'm a deno newbie, but I believe it should try to identify the encoding of the terminal. At least it should let me select the encoding.

**Comment by rafawalter** [comment]
That would be awesome!

My Windows prompt defaults to CP 850. Ending up with problems with Deno and prompting common names like "João" :-(

**Comment by rafawalter** [comment]
May be related to https://github.com/denoland/deno/issues/8239
